craft_parts/lifecycle_manager.py

```diff
@@ -110,21 +119,41 @@ def __init__(
         for name, spec in parts_data.items():
             part_list.append(_build_part(name, spec, project_dirs))
 
+        self._has_overlay = any(p.has_overlay for p in part_list)
+
+        # a base layer is mandatory if overlays are in use
+        if self._has_overlay:
+            if not base_layer_dir:
+                raise ValueError("using overlays and base_layer_dir not specified")
+            if not base_layer_hash:
+                raise ValueError("using overlays and base_layer hash not specified")
```

Comment:
The error message looks weird starting with "using overlays" (because that's not an error).

I propose to turn both messages around, in the line of "The base_layer_dir must be specified when using overlays.".

What do you think?

Reply:
Rephrased as suggested.